app/src/main/java/de/aurora/mggvertretungsplan/MainActivity.java

```diff
@@ -220,6 +221,68 @@ public boolean onOptionsItemSelected(MenuItem item) {
             case R.id.action_feedback:
                 launchCustomTabsIntent(color, getString(R.string.feedback_url));
                 break;
+            case R.id.action_share:
+                // Share
+                String shareText = "Vertretungsplan " + class_name;
+                String input = StorageUtilities.readFile(this);
+                try {
+                    JSONArray json = new JSONArray(input);
+                    // For every Week:
+                    for (int i = 0; i < json.length(); i++) {
+                        JSONObject dateData = json.getJSONObject(i);
+                        String Date = dateData.getString("date");
+                        String Week = dateData.getString("week");
+                        shareText = shareText + "\n" + Date + " Woche " + Week;
+                        JSONArray dateElements = dateData.getJSONArray("elements");
+                        // For every Element of the Week:
+                        for (int j = 0; j < dateElements.length(); j++){
```

Comment:
Waaaay too much work here. Use the timetable object (talked about that in the comment above). This is not bad, it's just too much work 🙈.